solidity/test/LinkEx_test.ts

```diff
@@ -1,24 +1,195 @@
+    const expectedAvg = Math.trunc((expected + expected2 + expected3) / 3)
+
+    context('when called by a stranger', () => {
+      it('reverts', async () => {
+        await h.assertActionThrows(async () => {
+          await contract.update(expected, {from: h.stranger})
+        })
+        const rate = await contract.currentRate()
+        assert.equal(rate, 0)
+      })
+    })
+
+    context('when called by an authorized node', () => {
+      beforeEach(async () => {
+        await contract.addOracle(h.oracleNode, {from: h.defaultAccount})
+      })
+
+      it('updates the rate', async () => {
+        await contract.update(expected, {from: h.oracleNode})
+        const historicRate = await contract.currentRate.call()
+        assert.equal(historicRate.toString(), expected.toString())
+      })
+    })
+
+    context('when updated recently by oracles', () => {
+      beforeEach(async () => {
+        await contract.addOracle(h.oracleNode, {from: h.defaultAccount})
+        await contract.addOracle(h.oracleNode2, {from: h.defaultAccount})
+        await contract.addOracle(h.oracleNode3, {from: h.defaultAccount})
+
+        await contract.update(expected, {from: h.oracleNode})
+        await contract.update(expected2, {from: h.oracleNode2})
+        await contract.update(expected3, {from: h.oracleNode3})
+      })
+
+      it('has an expected aggregate value', async () => {
+        const rate = await contract.currentRate()
+        assert.equal(rate, expectedAvg)
+      })
+
+      context('after adding more oracles', () => {
+        const expected4 = 8616460198
+        const expected5 = 8616460756
+        const newExpectedAvg = Math.trunc((expected + expected2 + expected3 + expected4 + expected5) / 5)
+
+        beforeEach(async () => {
+          await contract.addOracle(h.accounts[8], {from: h.defaultAccount})
+          await contract.addOracle(h.accounts[9], {from: h.defaultAccount})
+
+          await contract.update(expected4, {from: h.accounts[8]})
+          await contract.update(expected5, {from: h.accounts[9]})
+        })
+
+        it('the new oracles contribute to the average', async () => {
+          const rate = await contract.currentRate()
+          assert.equal(rate, newExpectedAvg)
+        })
+      })
+
+      context('after removing an oracle', () => {
+        const updated = 8616460198
+        const newExpectedAvg = Math.trunc((updated + expected2) / 2)
+
+        beforeEach(async () => {
+          await contract.removeOracle(h.oracleNode3, {from: h.defaultAccount})
+          await contract.update(updated, {from: h.oracleNode})
+        })
+
+        it('the removed oracles do not contribute to the average', async () => {
+          const rate = await contract.currentRate()
+          assert.equal(rate, newExpectedAvg)
+        })
+      })
+
+      context('when updated by an oracle after 25 blocks', () => {
+        beforeEach(async () => {
+          for (let i = 0; i < 25; i++) {
```

Comment:
Since we do this in multiple paces might be best to wrap as a helper e.g: `h.mineBlocks(25)` ?